docs/specifications/tests/DNSQueryAndResponsSpecification.md

```diff
@@ -0,0 +1,154 @@
+setting items specified below.
+
+A test case specification can refer to an *EDNS Query* with one or several
+changes to the setting items. Specifically, the test case specification must
+always specify query name and query type.
+
+|Setting Item          |Default value |Comment                   |
+|:---------------------|:-------|:-------------------------------|
+|ENDS                  | yes    | OPT record is included         |
+|EDNS UDP Message size | 512    |                                |
+|ENDS Extended RCODE   | no     |                                |
+|ENDS Version          | 0      |                                |
+|ENDS DO flag          | unset  |                                |
+|EDNS Z lag            | unset  |                                |
+|EDNS0 Option          | none   |                                |
+
+
+## Default setting in *DNSSEC Query*
+
+An *DNSSEC Query* inherit the default setting from an *EDNS Query* except for the
+setting items specified below.
+
+A test case specification can refer to an *DNSSEC Query* with one or several
+changes to the setting items. Specifically, the test case specification must
+always specify query name and query type.
+
+|Setting               |Default value |Comment                   |
+|:---------------------|:-------|:-------------------------------|
+|ENDS DO flag          | set    |                                |
+
+
+## Default handling of a *DNS Response*
+
+A *DNS Response* is a response to a *DNS Query*. Unless specified in the test
+case specification, the items in the response is handled as listed.
+
+|Response Item |Default handling    | Comment                         |
+|:-------------|:-------------------|:--------------------------------|
+|OpCode        | Must be "response" | Always "response" in a response |
+|QR flag       | Must be set        | Always set in a response        |
+|AA flag       | -                  | Defined in test case            |
+|TC flag       | Re-query over TCP   |                                 |
+|RD flag       | ignore             |                                 |
+|RA flag       | ignore             |                                 |
+|AD flag       | ignore             |                                 |
+|CD flag       | ignore             |                                 |
+|RCODE         | -                  | Defined in test case            |
+|Query name    | ignore             |                                 |
+|Query type    | ignore             |                                 |
+|ENDS          | ignore             |                                 |
+
+
+* Check against query name and query type is, by default, done against the values
+  in the query, not in the response.
+  
+* When fetching records from the answer section, these are the default criteria:
+  * Only records matching the query name and the query type are fetched. RRSIG
+    records, when available, meeting the next criterium are also fetched.
```

Comment:
Maybe remove the "when available" and the commas surrounding it? It makes the sentence more complex without helping understanding.

Reply:
Changed by putting an "Any" first in the sentence.